packages/framework/collabspaces/src/test/collabSpaces.spec.ts

```diff
@@ -309,6 +310,104 @@ describe("Temporal Collab Spaces", () => {
+
+			assert(rowMapSize === rows, "rowMapSize is incorrect");
+			assert(rowIndex === row + 1, "rowIndex from the actual matrix has to be offset by 1");
+			assert(colMapSize === cols, "colMapSize is incorrect");
+			assert(colIndex === col + 1, "colIndex is correct");
+		});
+
+		it("Reverse Mapping: Basic row adding test", async () => {
+			const collabSpace = await initialize();
+			const row = 5;
+			const col = 3;
+			const numberOfNewRows = 2;
+			// Insert rows to validate the reverse mappings are updated correctly
+			collabSpace.insertRows(1, numberOfNewRows);
+			collabSpace.setCell(row, col, {
+				value: 5,
+				type: CounterFactory.Type,
+			});
+			const { rowId, colId } = await getRowAndColIdsFromChannelId(collabSpace, row, col);
+			const { rowMapSize, colMapSize, rowIndex, colIndex } =
+				collabSpace.getReverseMapsDebugInfo(rowId, colId);
+			assert(rowMapSize === rows + numberOfNewRows, "rowMapSize is incorrect");
+			assert(rowIndex === row + 1, "rowIndex is correct");
+			assert(colMapSize === cols, "colMapSize is incorrect");
+			assert(colIndex === col + 1, "colIndex is correct");
+		});
+
+		it("Reverse Mapping: Basic row removing test", async () => {
+			const collabSpace = await initialize();
+			const row = 1;
+			const col = 3;
+
+			const { rowId: nextRowId, colId: nextColId } = await getRowAndColIdsFromChannelId(
+				collabSpace,
+				row + 1,
+				col,
+			);
+			collabSpace.removeRows(row, 1);
+
+			const { rowId, colId } = await getRowAndColIdsFromChannelId(collabSpace, row, col);
+			assert(nextRowId === rowId, "rowId after removal should be the same as nextRowId");
+			assert(nextColId === colId, "colId after removal should be different");
+
+			const { rowMapSize, colMapSize, rowIndex, colIndex } =
+				collabSpace.getReverseMapsDebugInfo(rowId, colId);
+
+			assert(rowMapSize === rows - 1, "rowMapSize is incorrect");
+			assert(rowIndex === row + 1, "rowIndex is correct");
+			assert(colMapSize === cols, "colMapSize is incorrect");
+			assert(colIndex === col + 1, "colIndex is correct");
+		});
+
+		it("Reverse Mapping: Basic col removing test", async () => {
+			const collabSpace = await initialize();
+			const row = 1;
+			const col = 3;
+			const columnsToBeRemoved = 1;
+			const { colId: nextColId } = await getRowAndColIdsFromChannelId(
+				collabSpace,
+				row,
+				col + 1,
+			);
+			collabSpace.removeCols(col, columnsToBeRemoved);
+			const { rowId, colId } = await getRowAndColIdsFromChannelId(collabSpace, row, col);
+
+			assert(nextColId === colId, "colId after removal should be the same as nextColId");
+
+			const { rowMapSize, colMapSize, rowIndex, colIndex } =
+				collabSpace.getReverseMapsDebugInfo(rowId, colId);
+			assert(rowMapSize === rows, "rowMapSize is incorrect");
+			assert(rowIndex === row + 1, "rowIndex is correct");
+			assert(colMapSize === cols - columnsToBeRemoved, "colMapSize is incorrect");
+			assert(colIndex === col + 1, "colIndex is correct");
```

Comment:
I'd suggest to have at least one test that deals with changes in another container, and validates that mappings are Ok in both containers after changes are synced. Including having concurrent changes (like insert a row in both), similar to what we have with ChannelOverwrite() (with parametrizing of having synchronization vs. not, and thus testing concurrency vs. non-concurrent change propagation)

Reply:
yes, I plan on doing it but I wanted to see if there is a better way to store and update the reverse mapping matrixes  for sure.